src/main/java/de/rub/nds/anvilcore/context/AnvilContext.java

```diff
@@ -43,9 +43,24 @@ public class AnvilContext {
     @JsonProperty("Score")
     private final ScoreContainer overallScoreContainer = new ScoreContainer();
 
+    /**
+     * A Map that keeps track of all the active test runs in the current context. The keys in this
+     * map are String objects representing unique IDs of test runs, and the values are AnvilTestRun
+     * objects representing the test runs themselves.
+     */
     private final Map<String, AnvilTestRun> activeTestRuns = new HashMap<>();
-    private final Map<TestResult, List<String>> resultTestMap = new HashMap<>();
-    private final Map<String, Boolean> finishedTests = new HashMap<>();
+    /**
+     * A Map that holds the test results. The keys are TestResult objects, and the values are
+     * List<String> objects representing the names of the tests.
+     */
+    private final Map<TestResult, List<AnvilTestRun>> resultsTestRuns =
+            new HashMap<TestResult, List<AnvilTestRun>>();
```

Comment:
This depends on how our report file should look like, which is created by AnvilReport. Currently we need this to include the failure details in the report, which I added in [this commit](https://github.com/tls-attacker/Anvil-Core/pull/31/commits/a2cabda43525291766a468ced0d2c7b0c8d70212).

I believe it's not worth utilizing a huge amount of RAM for this feature (and thus, remove it) but would like to hear your opinion on this.

Reply:
Aren't the failure details ultimately only strings which we could store in a map? This should be much more RAM efficient than keeping all State objects. Regardless, I think the average user won't benefit from the overview of all failure reasons at the end. Maybe we drop this part of the final summary entirely, though it is helpful for us to spot issues at glance.

Reply:
I have tried implementing it with a map like this in AnvilContext:

```java
/**
     * A map that holds the test case failure details for fully failed test cases. The keys are test case failure details
     * and the values are failure detail occurrence counters.
     */
    private final Map<String, Integer> fullyFailedTestCasesDetails = new HashMap<>();
```

but this quickly bloats the code:

It needs eventually two maps in AnvilContext. One for partially, one for fully failed results. This means you need two getter methods again and have to make sure that the failure details of the test cases are sorted correctly in the addTestResult method. In my opinion, it would only be worth bloating the code in this way if it would be of great benefit to us developers.

Translated with DeepL.com (free version)

Reply:
We don't need to differentiate between the failure reasons of fully failed and partially failed at this point. Both should be addressed by the developer or excluded as intentional non-compliances (though partially failed tests are more indicative of a bug).

Reply:
I implemented your change requests in commit [65c2ed8](https://github.com/tls-attacker/Anvil-Core/pull/31/commits/65c2ed8584a653335f4f813dc160261c941eeb96)